compiler/passes/normalize.cpp

```diff
@@ -276,6 +278,82 @@ static void transformLogicalShortCircuit()
   }
 }
 
+
+//
+// lowerReduceAssign(): lower the reduce= calls, verifying their correct use.
+//
+static Symbol* reduceIntentOp(BlockStmt* block, Symbol* reVar) {
+  // See markOuterVarsWithIntents() on how to decode byrefVars.
+  // Note that here we may have leading iter symbols, see getIterSymbols().
+  CallExpr* byrefVars = block->byrefVars;
+  bool leadingIterSymbols =
+    byrefVars->numActuals() >= 3 &&
+    toSymExpr(byrefVars->get(1))->var->hasFlag(FLAG_CHPL__ITER);
+
+  // The first actual (after the iter symbols when present); null if none.
+  Expr* markExpr = leadingIterSymbols ? byrefVars->get(3)->next :
+    byrefVars->argList.head;
+
+  while (markExpr) {
+    SymExpr* markSE = toSymExpr(markExpr);
+    SymExpr* outervarSE = toSymExpr(markSE->next);
+    markExpr = outervarSE->next;
+
+    if (isVarSymbol(markSE->var))
+      // this is a reduce intent
+      if (outervarSE->var == reVar)
+        // found reVar
+        return markSE->var;
+  }
+
+  // Did not see 'reVar' with a reduce intent.
+  return NULL;
+}
+static void lowerReduceAssign() {
+  forv_Vec(CallExpr, call, gCallExprs) {
+    if (call->isPrimitive(PRIM_REDUCE_ASSIGN)) {
+
+      // l.h.s. must be a single variable
+      if (SymExpr* lhsSE = toSymExpr(call->get(1))) {
+        Symbol* lhsVar = lhsSE->var;
+        // ... which is mentioned in a with clause with a reduce intent
+        Expr* curr = call->parentExpr;
+        BlockStmt* enclosingForall = NULL;
+        while (curr) {
+          if (BlockStmt* block = toBlockStmt(curr))
+            if (block->byrefVars) {
```

Comment:
I'm unfamilier with BlockStmt -> byrefVars but it looks like it is storing a PRIM_FORALL_LOOP in which case it really ought to be renamed. As it is I am confused why lowering reduce= needs to concern itself with any variable called "byrefVars".